### DISCUSSION TITLE
Button icon color - RGB light

### DISCUSSION THREAD

**Discussion by polypworld** [Blackymas/NSPanel_HA_Blueprint#551] [Ideas]
Hi,
If you paste the code below between lines 4827-4828, 5078-5079, 5329-5330, 5580-5581, the color of the icon will be adjusted to the color of the RGB lamp.
Have a nice day

'''
{%- elif repeat.item.entity is match "light." and current_entity_state == 'on' and state_attr(repeat.item.entity, "rgb_color") != None -%} {{ (state_attr(repeat.item.entity, "rgb_color")[0]|bitwise_and(248)*256) + (state_attr(repeat.item.entity, "rgb_color")[1]|bitwise_and(252)*8) + (state_attr(repeat.item.entity, "rgb_color")[2]|bitwise_and(248)/8)|int }}
'''

**Comment by edwardtfn** [comment]
> the color of the icon will be adjusted to the color of the RGB lamp

You mean the color of the icon in the button pages? I liked the idea.
What if the light color is similar to the background color, which won't be a rare situation? Any thoughts about that?

**Reply by polypworld** [reply]
Thank you, this is important info in my modified code. The quesrioms is correct, if the icon gets a frame, then it has a marked value...

**Comment by cr4shydlo** [comment]
Hello. I search for this mod ;) Where i need to add this code ?
